GoogleSignIn/Sources/GIDGoogleUser.m

```diff
@@ -201,4 +243,104 @@ - (void)encodeWithCoder:(NSCoder *)encoder {
 
 @end
 
+#if TARGET_OS_IOS && !TARGET_OS_MACCATALYST
+
+#pragma mark - GTMAppAuthFetcherAuthorizationEMMChainedDelegate
+
+// The specialized GTMAppAuthFetcherAuthorization delegate that handles potential EMM error
+// responses.
+@interface GTMAppAuthFetcherAuthorizationEMMChainedDelegate : NSObject
+
+// Initializes with chained delegate and selector.
+- (instancetype)initWithDelegate:(id)delegate selector:(SEL)selector;
+
+// The callback method for GTMAppAuthFetcherAuthorization to invoke.
+- (void)authentication:(GTMAppAuthFetcherAuthorization *)auth
+               request:(NSMutableURLRequest *)request
+     finishedWithError:(nullable NSError *)error;
+
+@end
+
+@implementation GTMAppAuthFetcherAuthorizationEMMChainedDelegate {
+  // We use a weak reference here to match GTMAppAuthFetcherAuthorization.
+  __weak id _delegate;
+  SEL _selector;
+  // We need to maintain a reference to the chained delegate because GTMAppAuthFetcherAuthorization
+  // only keeps a weak reference.
+  GTMAppAuthFetcherAuthorizationEMMChainedDelegate *_retained_self;
+}
+
+- (instancetype)initWithDelegate:(id)delegate selector:(SEL)selector {
+  self = [super init];
+  if (self) {
+    _delegate = delegate;
+    _selector = selector;
+    _retained_self = self;
+  }
+  return self;
+}
+
+- (void)authentication:(GTMAppAuthFetcherAuthorization *)auth
+               request:(NSMutableURLRequest *)request
+     finishedWithError:(nullable NSError *)error {
+  [GIDAuthentication handleTokenFetchEMMError:error completion:^(NSError *_Nullable error) {
+    if (!self->_delegate || !self->_selector) {
```

Comment:
I mean why are you using `self->_delegate` as opposed to `self.delegate`?

Reply:
Because it is an ivar rather than a property.

Reply:
Sorry I wasn't being more clear. I've written an issue to track the sort of improvement I have in mind. This work can be taken up in a separate project since I realize this PRs is moving preexisting code and the pattern I'm critiquing here is used throughout the project.

#232

Reply:
Thanks for creating this issue. We can fix them in another PR altogether.